.github/workflows/continuous-integration-workflow.yml

```diff
@@ -7,12 +7,42 @@ on:
       - 'README.md'
 
 jobs:
-  build:
+  build-and-test:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: build docker container 
       run: docker build --target test --tag todo-app:test .
     - name: run docker tests
-      run: docker run todo-app:test
\ No newline at end of file
+      run: docker run todo-app:test
+      
+  build-and-push-docker-image:
+    name: Docker Build, Tag & Publish
+    needs: build-and-test
+    if: ${{ github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout Code
+        uses: actions/checkout@v1
+        id: git_checkout
+
+      - name: Build, Tag & Publish Image to DockerHub
+        uses: dinushchathurya/build-tag-push-action@v1.2.1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          context: .
+          file: Dockerfile
```

Comment:
I notice this doesn't specify a `target` stage for your Dockerfile - that means it's actually defaulting to the last defined stage which for you is the test stage - so if I run the following:
```
 docker run -it --env-file .env emilie03/todo-app:exercise-9
```
Then it actually runs your tests rather than your prod image, which I don't think is your intention here!

Probably easiest is to either:
* Move your production stage to be the last one in your Dockerfile, probably worth adding a comment to explain why too
* Switch off this action & just run the relevant commands manually

Although if you fancy raising a PR to the relevant action to make target an option that would score some bonus points 😄

Reply:
I deleted the resource on Azure when I was done as you said to delete things when you were finished 😢
I'll create it again when I have a sec :)